components/core/tests/test-clp_s-search.cpp

```
@@ -0,0 +1,193 @@
+            {R"aa(msg: "*Abc123*")aa", {1, 2, 3, 5, 6}},
+            {R"aa(arr.b > 1000)aa", {7, 8}}
+    };
+    auto structurize_arrays = GENERATE(true, false);
+    auto single_file_archive = GENERATE(true, false);
+
+    TestOutputCleaner const test_cleanup{{std::string{cTestSearchArchiveDirectory}}};
+
+    compress(structurize_arrays, single_file_archive);
+
+    for (auto const& [query, expected_results] : queries_and_results) {
+        search(query, false, expected_results);
```

cppcheck (2.10-2)

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)